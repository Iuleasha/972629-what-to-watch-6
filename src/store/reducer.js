import {AuthorizationStatus, DEFAULT_GENRE} from '../constant';
import {createGenreArray} from '../filter/filter';
import {ActionType} from './action';

const initialState = {
  genre: DEFAULT_GENRE,
  genres: [],
  films: [],
  authorizationStatus: AuthorizationStatus.NO_AUTH,
  isDataLoaded: false,
  comments: {},
  preview: null,
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.SET_GENRE:
      return {
        ...state,
        genre: action.payload,
      };

    case ActionType.LOAD_SUCCESS:
      return {
        ...state,
        genres: createGenreArray(action.payload),
        films: action.payload,
        isDataLoaded: true,
      };

    case ActionType.LOAD_PROMO_FILM:
      return {
        ...state,
        preview: action.payload,
      };

    case ActionType.REQUIRED_AUTHORIZATION:
      return {
        ...state,
        authorizationStatus: action.payload,
      };

    case ActionType.SET_USER:
      return {
        ...state,
        user: action.payload,
      };

    case ActionType.ADD_REVIEW:
      const {comments} = state;

      return {
        ...state,
        comments: {...comments, [action.payload.id]: action.payload.comments},
      };

    case ActionType.LOAD_FAVORITE:
      return {
        ...state,
        favorite: action.payload,
      };

    case ActionType.SWITCH_FAVORITE_STATUS:
      if (state.preview && state.preview.id === action.payload.id) {
        state.preview = action.payload;
      }

      state.films = state.films.map((item) => item.id === action.payload.id ?
        action.payload : item
      );

      return {
        ...state,
        favorite: state.films.filter(({isFavorite}) => isFavorite),
      };
  }

  return state;
};


export {reducer};
