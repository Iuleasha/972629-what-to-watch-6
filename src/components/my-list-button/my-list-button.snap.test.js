
import {configureStore} from '@reduxjs/toolkit';
import {createMemoryHistory} from 'history';
import React from 'react';
import {render} from '@testing-library/react';
import {Provider} from 'react-redux';
import {Router} from 'react-router-dom';
import reducer from '../../store/root-reducer';
import MyListButton from './my-list-button';

const store = configureStore({reducer});

describe(`Test MyListButton`, () => {
  const history = createMemoryHistory();

  test(`Should render correctly if isFavorite false`, () => {
    const {container} = render(
        <Provider store={store}>
          <Router history={history}>
            <MyListButton isFavorite={false} id={1}/>
          </Router>
        </Provider>);
    expect(container).toMatchSnapshot();
  });

  test(`Should render correctly if isFavorite true`, () => {
    const {container} = render(
        <Provider store={store}>
          <Router history={history}>
            <MyListButton isFavorite={true} id={1}/>
          </Router>
        </Provider>);
    expect(container).toMatchSnapshot();
  });
});
