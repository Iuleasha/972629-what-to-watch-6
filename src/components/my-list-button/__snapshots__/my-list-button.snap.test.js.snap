// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test MyListButton Should render correctly if isFavorite false 1`] = `
<div>
  <button
    class="btn movie-card__button"
    type="button"
  >
    <svg
      height="20"
      viewBox="0 0 19 20"
      width="19"
    >
      <use
        xlink:href="#add"
      />
    </svg>
    <span>
      My list
    </span>
  </button>
</div>
`;

exports[`Test MyListButton Should render correctly if isFavorite true 1`] = `
<div>
  <button
    class="btn movie-card__button"
    type="button"
  >
    <svg
      height="20"
      viewBox="0 0 19 20"
      width="19"
    >
      <use
        xlink:href="#in-list"
      />
    </svg>
    <span>
      My list
    </span>
  </button>
</div>
`;
